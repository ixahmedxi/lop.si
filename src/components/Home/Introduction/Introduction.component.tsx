/* @jsx jsx */
import { Box, jsx, SxStyleProp, Text } from 'theme-ui'

const Wrapper: SxStyleProp = {
  margin: '0 auto',
  width: ['100%', '100%', '100%', '70%'],
  textAlign: 'center'
}

const Title: SxStyleProp = {
  color: 'primary',
  fontSize: ['20px', '20px', 3],
  fontWeight: 'heading',
  pb: [1, 1, 2]
}

const Description: SxStyleProp = {
  lineHeight: 1,
  fontSize: [0, 0, 1]
}

export const Introduction: React.FC = () => {
  return (
    <Box sx={Wrapper} data-testid="introduction">
      <Text as="h1" sx={Title}>
        Create professional looking urls
      </Text>
      <Text as="p" sx={Description}>
        lop.si helps you transform long and unmanageable urls into short alternatives that look more
        professional wherever used.
      </Text>
    </Box>
  )
}
