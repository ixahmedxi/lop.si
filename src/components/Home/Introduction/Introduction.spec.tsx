import { render } from 'test-utils'
import { Introduction } from './Introduction.component'

describe('Introduction Component', () => {
  it('matches snapshot', () => {
    const { container } = render(<Introduction />)
    expect(container).toMatchSnapshot()
  })
})
