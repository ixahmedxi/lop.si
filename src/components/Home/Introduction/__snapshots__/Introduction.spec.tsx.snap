// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction Component matches snapshot 1`] = `
.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 0 auto;
  width: 100%;
  text-align: center;
}

@media screen and (min-width:576px) {
  .emotion-2 {
    width: 100%;
  }
}

@media screen and (min-width:768px) {
  .emotion-2 {
    width: 100%;
  }
}

@media screen and (min-width:992px) {
  .emotion-2 {
    width: 70%;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-primary,#8093f2);
  font-size: 20px;
  font-weight: 700;
  padding-bottom: 8px;
}

@media screen and (min-width:576px) {
  .emotion-0 {
    font-size: 20px;
    padding-bottom: 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    font-size: 32px;
    padding-bottom: 16px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  line-height: 1.8;
  font-size: 12px;
}

@media screen and (min-width:576px) {
  .emotion-1 {
    font-size: 12px;
  }
}

@media screen and (min-width:768px) {
  .emotion-1 {
    font-size: 16px;
  }
}

<div>
  <div
    class="emotion-2"
    data-testid="introduction"
  >
    <h1
      class="emotion-0"
    >
      Create professional looking urls
    </h1>
    <p
      class="emotion-1"
    >
      lop.si helps you transform long and unmanageable urls into short alternatives that look more professional wherever used.
    </p>
  </div>
</div>
`;
